tests/spec/core/dfn-spec.js

```diff
@@ -329,4 +329,218 @@ describe("Core — Definitions", () => {
+      const dfn = doc.querySelector("#element-state dfn");
+      expect(dfn.dataset.dfnFor).toBe("input");
+      expect(dfn.dataset.dfnType).toBe("element-state");
+      expect(dfn.dataset.export).toBe("");
+
+      // Check validation error
+      const errors = doc.respec.errors.filter(err => err.plugin === "core/dfn");
+      // missing data-dfn-for and invalid name
+      expect(errors).toHaveSize(2);
+    });
+
+    it("handles events", async () => {
+      const body = `
+        <section>
+          <h2>Events</h2>
+          <p id="events">
+            <dfn class="event">DOMContentLoaded</dfn>
+          </p>
+          <p id="events-bad">
+            <dfn class="event">-event</dfn>
+          </p>
+        </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#events dfn");
+      expect(dfn.dataset.dfnType).toBe("event");
+      expect(dfn.dataset.export).toBe("");
+
+      // Check validation error
+      const errors = doc.respec.errors.filter(err => err.plugin === "core/dfn");
+      expect(errors).toHaveSize(1);
+      expect(errors[0].message).toContain("-event");
+    });
+
+    it("handles http-headers", async () => {
+      const body = `
+        <section>
+          <h2>HTTP Header</h2>
+          <p id="http-header">
+            <dfn class="http-header">Some-Header</dfn>
+          </p>
+        </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#http-header dfn");
+      expect(dfn.dataset.dfnType).toBe("http-header");
+      expect(dfn.dataset.export).toBe("");
+    });
+
+    it("handles schemes", async () => {
+      const body = `
+        <section>
+          <h2>Scheme</h2>
+          <p id="http-header">
+            <dfn class="scheme">scheme</dfn>
+          </p>
+        </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#http-header dfn");
+      expect(dfn.dataset.dfnType).toBe("scheme");
+      expect(dfn.dataset.export).toBe("");
+    });
+
+    it("handles media types", async () => {
+      const body = `
+        <section>
+          <h2>HTTP Header</h2>
+          <p id="http-header">
+            <dfn class="media-type">application/something-something</dfn>
+          </p>
+          <p id="http-header-bad">
+            <dfn class="media-type">bad type</dfn>
+        </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#http-header dfn");
+      expect(dfn.dataset.dfnType).toBe("media-type");
+      expect(dfn.dataset.export).toBe("");
+
+      // Check validation error
+      const errors = doc.respec.errors.filter(err => err.plugin === "core/dfn");
+      expect(errors).toHaveSize(1);
+    });
+  });
```

Comment:
Let's also add some tests on what happens if we provided multiple classes (constructive as well as clashing).

Reply:
Yeah, I deliberately left this case undefined. I wanted to get some experience with them before handling that particular case (i.e. first one wins right now). I can at least test that.

Reply:
Ok, cool. This we backwards... it was giving the names in the Map priority.